File: 0014-longest-common-prefix/0014-longest-common-prefix.scala
object Solution {
    def longestCommonPrefix(strs: Array[String]): String = {
        if (strs.isEmpty) return ""

        var pre = strs(0)
        var i = 1
        while (i < strs.length) {
          while (!strs(i).startsWith(pre)) {
            pre = pre.substring(0, pre.length - 1)
          }
          i += 1
        }
        pre
    }
}

File: 0014-longest-common-prefix/0014-longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        String pre = strs[0];
        int i =1;
        while(i<strs.length){
            if(strs[i].startsWith(pre)){
                i++;
            }
            else{
                pre = pre.substring(0, pre.length()-1);
            }
        }
        return pre;
    }
}